OpenRA.Mods.Common/Traits/BotModules/McvExpansionManagerBotModule.cs

```diff
@@ -0,0 +1,767 @@
+					i % resourceMapIndicesColumnCount * resourceIndiceStride + (resourceIndiceStride >> 1),
+					i / resourceMapIndicesRowCount * resourceIndiceStride + (resourceIndiceStride >> 1)).ToCPos(map), 0)).Shuffle(world.LocalRandom).ToArray();
+				for (var i = 0; i < resourceMapIndices.Length; i++)
+					UpdateResourceMap(i);
+
+				pathDistanceSquareFactor = resourceMapIndicesColumnCount * resourceMapIndicesColumnCount + resourceMapIndicesRowCount * resourceMapIndicesRowCount;
+			}
+		}
+
+		void SwitchExpansionMode(BotMcvExpansionMode nextMode)
+		{
+			mcvExpansionMode = nextMode;
+			switch (nextMode)
+			{
+				case BotMcvExpansionMode.CheckResourceCreator:
+					mcvDeploymentMinDeployRadius = Info.CRmodeMinDeployRadius;
+					mcvDeploymentMaxDeployRadius = Info.CRmodeMaxDeployRadius;
+					mcvDeploymentTryMaintainRange = Info.CRCmodeTryMaintainRange;
+					break;
+
+				case BotMcvExpansionMode.CheckResource:
+					mcvDeploymentMinDeployRadius = Info.CBmodeMinDeployRadius;
+					mcvDeploymentMaxDeployRadius = Info.CBmodeMaxDeployRadius;
+					mcvDeploymentTryMaintainRange = Info.CRmodeTryMaintainRange;
+					break;
+
+				case BotMcvExpansionMode.CheckBase:
+					mcvDeploymentMinDeployRadius = Info.CBmodeMinDeployRadius;
+					mcvDeploymentMaxDeployRadius = Info.CBmodeMaxDeployRadius;
+					mcvDeploymentTryMaintainRange = (Info.CBmodeMaxDeployRadius + Info.CBmodeMinDeployRadius) >> 1;
+					break;
+
+				default:
+					break;
+			}
+		}
+
+		void FindBadDeploySpot(CPos? failedSpot)
+		{
+			lastFailedExpandSpot = failedSpot.Value;
+			if (++failedAttempts >= maxFailedAttempts)
+			{
+				failedAttempts = 0;
+				switch (mcvExpansionMode)
+				{
+					case BotMcvExpansionMode.CheckResourceCreator:
+						SwitchExpansionMode(BotMcvExpansionMode.CheckResource);
+						break;
+
+					case BotMcvExpansionMode.CheckResource:
+						SwitchExpansionMode(BotMcvExpansionMode.CheckBase);
+						break;
+
+					case BotMcvExpansionMode.CheckBase:
+						SwitchExpansionMode(BotMcvExpansionMode.CheckResourceCreator);
+						maxFailedAttempts = NegativeMaxFailedAttempts;
+						break;
+				}
+			}
+		}
+
+		void FindGoodDeploySpot()
+		{
+			lastFailedExpandSpot = null;
+			if (--failedAttempts <= -maxFailedAttempts)
+			{
+				maxFailedAttempts = PositiveMaxFailedAttempts;
+				switch (mcvExpansionMode)
+				{
+					case BotMcvExpansionMode.CheckResourceCreator:
+						failedAttempts = -maxFailedAttempts;
+						break;
+
+					case BotMcvExpansionMode.CheckBase:
+						failedAttempts = maxFailedAttempts - 1;
+						SwitchExpansionMode(BotMcvExpansionMode.CheckResource);
+						break;
+
+					case BotMcvExpansionMode.CheckResource:
+						failedAttempts = maxFailedAttempts - 1;
+						SwitchExpansionMode(BotMcvExpansionMode.CheckResourceCreator);
+						break;
+				}
+			}
+		}
+
+		public (CPos? ExpandLocation, int Attraction) GetExpandCenter(CPos mcv_loc, bool allowfallback)
```

Comment:
```suggestion
		public (CPos? ExpandLocation, int Attraction) GetExpansionCenter(CPos mcvLoc, bool allowfallback)
```

OpenRA convention disctates camel case. Let's also use "Expansion" terminology as that is what players already use

Reply:
fixed